indico/modules/events/client/js/creation.js

```diff
@@ -58,5 +76,209 @@
+
+            $('#event-creation-location_data').on('change', function() {
+                roomData = JSON.parse($(this).val());
+                if (previousRoomId !== roomData['room_id']) {
+                    updateAvailability(true);
+                }
+            });
+
+            $('#event-creation-start_dt-date').on('change', function() {
+                const startDate = $(this).val();
+                const startTime = $('#event-creation-start_dt-time').val();
+                const endDate = $('#event-creation-end_dt-date').val();
+                const endTime = $('#event-creation-end_dt-time').val();
+                startDt = moment(`${startDate} ${startTime}`, 'DD/MM/YYYY HH:mm');
+                endDt = moment(`${endDate} ${endTime}`, 'DD/MM/YYYY HH:mm');
+                // workaround for automatic end date update if start date is after end date
+                if (endDt.isBefore(startDt)) {
+                    endDt = moment(`${startDate} ${endTime}`, 'DD/MM/YYYY HH:mm');
+                }
+                updateAvailability();
+            });
+
+            $('#event-creation-start_dt-time').on('change', function() {
+                const startDate = $('#event-creation-start_dt-date').val();
+                const startTime = $('#event-creation-start_dt-time').val();
+                startDt = moment(`${startDate} ${startTime}`, 'DD/MM/YYYY HH:mm');
+                updateAvailability();
+            });
+
+            $('#event-creation-end_dt-date').on('change', function() {
+                const endDate = $(this).val();
+                const endTime = $('#event-creation-end_dt-time').val();
+                endDt = moment(`${endDate} ${endTime}`, 'DD/MM/YYYY HH:mm');
+                updateAvailability();
+            });
+
+            $('#event-creation-end_dt-time').on('change', function() {
+                const endDate = $('#event-creation-end_dt-date').val();
+                const endTime = $(this).val();
+                endDt = moment(`${endDate} ${endTime}`, 'DD/MM/YYYY HH:mm');
+                updateAvailability();
+            });
+
+            $('#event-creation-occurrences').on('change', function() {
+                const occurrences = JSON.parse($(this).val());
+                if (occurrences.length === 1) {
+                    setLectureTimes(occurrences[0]);
+                    multipleOccurrences = false;
+                } else {
+                    multipleOccurrences = true;
+                }
+                updateAvailability();
+            });
+
+            $('#event-creation-timezone').on('change', function() {
+                timezone = ($(this).val());
+                updateAvailability();
+            });
+
+            $('#event-creation-category').on('change', function() {
+                category = JSON.parse($(this).val());
+                updateAvailability();
+            });
+
+            $('#create-booking').on('change', function() {
```

Comment:
you repeat this code 3 times below, why not add a class for all these items and bind on that?

Reply:
it just seemed like a better idea, than adding a class parameter to switch macro

Reply:
you can also use `#foo, #bar, #...` in a single `$(...)` call

Reply:
thats a good idea